### PR TITLE
Now Water sources will also be created on top of existing water sources.

### LINKED ISSUE #2097: Water does not regenerate without a block underneath it.
This was changed in one of the recent patches, so it now regenerates with just water underneath it.

### PR DESCRIPTION
**Pull request by bibo38**
See also on http://minecraft.gamepedia.com/Water#Source_blocks
Fixes #2097

### CODE REVIEW

File: src/Simulator/FloodyFluidSimulator.cpp [hunk 1]
```diff
@@ -111,9 +111,11 @@ void cFloodyFluidSimulator::SimulateBlock(cChunk * a_Chunk, int a_RelX, int a_Re
 		
 		// If source creation is on, check for it here:
 		if (
-			(m_NumNeighborsForSource > 0) &&  // Source creation is on
-			(MyMeta == m_Falloff) &&          // Only exactly one block away from a source (fast bail-out)
-			!IsPassableForFluid(Below) &&     // Only exactly 1 block deep
+			(m_NumNeighborsForSource > 0) &&    // Source creation is on
+			(MyMeta == m_Falloff) &&            // Only exactly one block away from a source (fast bail-out)
+				(!IsPassableForFluid(Below) ||  // Only exactly 1 block deep
+				(Below == E_BLOCK_STATIONARY_WATER)  // Or a source block underneath
```
Comment (worktycho) [inline src/Simulator/FloodyFluidSimulator.cpp]: How does this work for lava?

Comment (bibo38) [inline src/Simulator/FloodyFluidSimulator.cpp]: Do you mean, when you activate Lava source block createion?
It wasn't defined, but I see your point. I've overseen the m_StationaryFluid variable. Will be fixed.

**Comment by bibo38** [comment]
Now it shold work for all fluid sources, if e.g. lava source creation is enabled.

File: src/Simulator/FloodyFluidSimulator.cpp [hunk 2]
```diff
@@ -111,9 +111,11 @@ void cFloodyFluidSimulator::SimulateBlock(cChunk * a_Chunk, int a_RelX, int a_Re
 		
 		// If source creation is on, check for it here:
 		if (
-			(m_NumNeighborsForSource > 0) &&  // Source creation is on
-			(MyMeta == m_Falloff) &&          // Only exactly one block away from a source (fast bail-out)
-			!IsPassableForFluid(Below) &&     // Only exactly 1 block deep
+			(m_NumNeighborsForSource > 0) &&    // Source creation is on
+			(MyMeta == m_Falloff) &&            // Only exactly one block away from a source (fast bail-out)
+				(!IsPassableForFluid(Below) ||  // Only exactly 1 block deep
```
Comment (tigerw) [inline src/Simulator/FloodyFluidSimulator.cpp]: Dis be some funky formatting, suggestion:
```cpp
(MyMeta == m_FallOff) &&
(
	!IsPassable() ||
	(Below == m_Stationary)
) &&
CheckNeighbours()
```

Comment (bibo38) [inline src/Simulator/FloodyFluidSimulator.cpp]: Seems legit. I first tried with adding a space before (Below == m_Stationary), but that failed in the style-checker.

**Comment by bibo38** [comment]
Now this PR should be mergeable.

**Pull request merged**